test/lock/optimistic_lock_test.cpp

```diff
@@ -0,0 +1,449 @@
+/*
+ * Copyright 2021 Database Group, Nagoya University
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "lock/optimistic_lock.hpp"
+
+#include <chrono>
+#include <future>
+#include <thread>
+#include <vector>
+
+#include "common.hpp"
+#include "gtest/gtest.h"
```

Comment:
前は順番適当にしていたんですが，自作プログラムをなるべく後で読むように最近は

1. 外部ライブラリ（`external libraries`），
2. 組織内ライブラリ（`organization libraries`），
3. ローカルソース（`local sources`）

の順番で読むようにしています．

```suggestion
// external libraries
#include "gtest/gtest.h"

// local sorces
#include "common.hpp"
```